tests/test_device_tracker.py

```diff
@@ -0,0 +1,204 @@
+        """Assert gets assigned to device"""
+        dev = qml.device("default.qubit", wires=2)
+
+        tracker = Tracker(dev=dev)
+
+        assert id(dev.tracker) == id(tracker)
+
+    def test_incompatible_device_assignment(self):
+        """Assert exception raised when `supports_tracker` not True"""
+
+        class TempDevice:
+            short_name = "temp"
+
+            def capabilities(self):
+                return dict()
+
+        temp = TempDevice()
+
+        with pytest.raises(Exception, match=r"Device 'temp' does not support device tracking"):
+            Tracker(dev=temp)
+
+    def test_reset(self):
+        """Assert reset empties totals and history"""
+
+        tracker = Tracker()
+
+        tracker.totals = {"a": 1}
+        tracker.history = {"a": [1]}
+        tracker.latest = {"a": 1}
+
+        tracker.reset()
+
+        assert tracker.totals == dict()
+        assert tracker.history == dict()
+        assert tracker.latest == dict()
+
+    def test_enter_and_exit(self):
+        """Assert entering and exit work as expected"""
+
+        tracker = Tracker()
+        tracker.totals = {"a": 1}
+        tracker.history = {"a": [1]}
+        tracker.latest = {"a": 1}
+
+        returned = tracker.__enter__()
+
+        assert id(tracker) == id(returned)
+        assert tracker.tracking == True
+
+        assert tracker.totals == dict()
+        assert tracker.history == dict()
+        assert tracker.latest == dict()
+
+        tracker.__exit__(1, 1, 1)
+
+        assert tracker.tracking == False
+
+    def test_context(self):
+        """Assert works with runtime context"""
+
+        with Tracker() as tracker:
+            assert isinstance(tracker, Tracker)
+            assert tracker.tracking == True
+
+        assert tracker.tracking == False
+
+    def test_update(self):
+        """Checks update stores to history and totals"""
+
+        tracker = Tracker()
+
+        tracker.update(a=1, b="b", c=None)
+        tracker.update(a=2, b="b2", c=1)
+
+        assert tracker.history == {"a": [1, 2], "b": ["b", "b2"], "c": [None, 1]}
+
+        assert tracker.totals == {"a": 3, "c": 1}
+
+        assert tracker.latest == {"a": 2, "b": "b2", "c": 1}
+
+    def test_record_callback(self, mocker):
+        class callback_wrapper:
+            @staticmethod
+            def callback(totals, history, latest):
+                pass
+
+        wrapper = callback_wrapper()
+        spy = mocker.spy(wrapper, "callback")
+
+        tracker = Tracker(callback=wrapper.callback)
+
+        tracker.totals = {"a": 1, "b": 2}
+        tracker.history = {"a": [1], "b": [1, 1]}
+        tracker.latest = {"a": 1, "b": 1}
+
+        tracker.record()
+
+        _, kwargs_called = spy.call_args_list[-1]
+
+        assert kwargs_called["totals"] == tracker.totals
+        assert kwargs_called["history"] == tracker.history
+        assert kwargs_called["latest"] == tracker.latest
+
+
+class TestDefaultTrackerIntegration:
```

Comment:
Though do I need these integration tests here if I have similar tests in the device suite?